### ISSUE TITLE
[image_picker] iOS 14 front camera shows green screen when flash mode is off 

### ISSUE THREAD

**Issue by CoderJava** [flutter/flutter#71131]
## Steps to Reproduce
1. Use `image_picker` to take picture from camera
2. By default, it will show rear camera
3. Switch to front camera and set flash light off
4. Take picture and the result is green screen

<img src="https://user-images.githubusercontent.com/17062085/100053519-ca3b1e00-2e52-11eb-8a44-97ca79951b31.jpeg" height="450" />

<details>
<summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel beta, 1.23.0-18.1.pre, on Mac OS X 10.15.7 19H2 x86_64, locale en-EC)
    • Flutter version 1.23.0-18.1.pre at /Users/yudisetiawan/fvm/versions/beta
    • Framework revision 198df796aa (6 weeks ago), 2020-10-15 12:04:33 -0700
    • Engine revision 1d12d82d9c
    • Dart version 2.11.0 (build 2.11.0-213.1.beta)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /Users/yudisetiawan/Library/Android/sdk
    • Platform android-30, build-tools 30.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.0.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.0.1, Build version 12A7300
    • CocoaPods version 1.10.0.rc.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] IntelliJ IDEA Community Edition (version 2020.2.3)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.50.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.16.0

[✓] Connected device (2 available)
    • Web Server (web) • web-server • web-javascript • Flutter Tools
    • Chrome (web)     • chrome     • web-javascript • Google Chrome 86.0.4240.198

• No issues found!
```
</details>

<details>
<summary>flutter crash logs</summary>

<code>
[Camera] Failed to read exposureBiasesByMode dictionary: Error Domain=NSCocoaErrorDomain Code=4864 "*** -[NSKeyedUnarchiver _initForReadingFromData:error:throwLegacyExceptions:]: data is NULL" UserInfo={NSDebugDescription=*** -[NSKeyedUnarchiver _initForReadingFromData:error:throwLegacyExceptions:]: data is NULL}
</code>
</details>

**Comment by darshankawar** [comment]
@CoderJava 
I tried the official `image_picker` plugin [example](https://github.com/flutter/plugins/tree/master/packages/image_picker/image_picker/example) using latest beta on iphone 6s (OS 14) and followed your steps to replicate, but didn't see the green screen. I was able to take a picture properly without any issues.

![IMG-0487](https://user-images.githubusercontent.com/67046386/100068900-1f306180-2e5e-11eb-8d19-b712d6dd73da.PNG)
![IMG-0488](https://user-images.githubusercontent.com/67046386/100068923-248dac00-2e5e-11eb-9858-67314bb309b2.PNG)


<details>
<summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel beta, 1.24.0-10.2.pre, on Mac OS X 10.15.4 19E2269
    darwin-x64, locale en-IN)
    • Flutter version 1.24.0-10.2.pre at /Users/dhs/documents/Fluttersdk/flutter
    • Framework revision 022b333a08 (6 days ago), 2020-11-18 11:35:09 -0800
    • Engine revision 07c1eed46b
    • Dart version 2.12.0 (build 2.12.0-29.10.beta)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.0)
    • Android SDK at /Users/dhs/Library/Android/sdk
    • Platform android-30, build-tools 30.0.0
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.0.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.0.1, Build version 12A7300
    • CocoaPods version 1.9.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 46.0.2
    • Dart plugin version 193.7361
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.50.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.15.1

[✓] Connected device (4 available)
    • Darshan's iphone (mobile)           •
      21150b119064aecc249dfcfe05e259197461ce23 • ios            • iOS 14.0
    • iPhone SE (2nd generation) (mobile) • 6C85835D-FBFD-4AB3-8DE8-B4FAD35E5367
      • ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-0 (simulator)
    • Web Server (web)                    • web-server
      • web-javascript • Flutter Tools
    • Chrome (web)                        • chrome
      • web-javascript • Google Chrome 87.0.4280.67

• No issues found!


```

</details>

Can you upgrade to latest beta and try again to see if you still get the same error ?
Thanks.

**Comment by CoderJava** [comment]
> @CoderJava
> I tried the official `image_picker` plugin [example](https://github.com/flutter/plugins/tree/master/packages/image_picker/image_picker/example) using latest beta on iphone 6s (OS 14) and followed your steps to replicate, but didn't see the green screen. I was able to take a picture properly without any issues.
> 
> ![IMG-0487](https://user-images.githubusercontent.com/67046386/100068900-1f306180-2e5e-11eb-8d19-b712d6dd73da.PNG)
> ![IMG-0488](https://user-images.githubusercontent.com/67046386/100068923-248dac00-2e5e-11eb-9858-67314bb309b2.PNG)
> 
> flutter doctor -v
> Can you upgrade to latest beta and try again to see if you still get the same error ?
> Thanks.

Did your flash mode is off atau on?

**Comment by CoderJava** [comment]
@darshankawar I'm already upgraded to latest beta. But I'm still have the same error.

[Video](https://youtu.be/FsPG7iIUFjg)

**Comment by darshankawar** [comment]
@CoderJava 
I had flash mode `off` when I tried. Can you provide a minimal reproducible code sample that shows the issue ?
You may also try the official plugin example in your environment and see if you similar error or not.
Thanks.

**Comment by CoderJava** [comment]
@darshankawar I'm already try example code. But I'm still have the same error. This error only happen when flash mode is off.

**Comment by darshankawar** [comment]
@CoderJava 
What iOS device are you using ? I tried on iPhone 6s and didn't see the issue.

**Comment by CoderJava** [comment]
@darshankawar I tried on iPhone 6s too.

**Comment by darshankawar** [comment]
@CoderJava Strange that earlier when I tried on latest beta, I didn't get the issue, but now I switched to latest stable and tried, I am able to replicate it now. Switched back to beta and seeing the issue. I think there's pattern to it while switching off the flash mode. Anyways, thanks for reporting this.

![IMG-0489](https://user-images.githubusercontent.com/67046386/100103374-cfb45a80-2e8a-11eb-9121-81b8e3fcaa92.PNG)

<details>
<summary>flutter doctor -v</summary>

```

[✓] Flutter (Channel beta, 1.24.0-10.2.pre, on Mac OS X 10.15.4 19E2269
    darwin-x64, locale en-IN)
    • Flutter version 1.24.0-10.2.pre at /Users/dhs/documents/Fluttersdk/flutter
    • Framework revision 022b333a08 (6 days ago), 2020-11-18 11:35:09 -0800
    • Engine revision 07c1eed46b
    • Dart version 2.12.0 (build 2.12.0-29.10.beta)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.0)
    • Android SDK at /Users/dhs/Library/Android/sdk
    • Platform android-30, build-tools 30.0.0
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.0.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.0.1, Build version 12A7300
    • CocoaPods version 1.9.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 46.0.2
    • Dart plugin version 193.7361
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.50.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.15.1

[✓] Connected device (3 available)
    • Darshan's iphone (mobile) • 21150b119064aecc249dfcfe05e259197461ce23 • ios
      • iOS 14.0
    • Web Server (web)          • web-server                               •
      web-javascript • Flutter Tools
    • Chrome (web)              • chrome                                   •
      web-javascript • Google Chrome 87.0.4280.67

• No issues found!

```

</details>

**Comment by TahaTesser** [comment]
Hi @CoderJava 
Tried to reproduce it again using `0.7.0-nullsafety`, with flash off, fresh install, no green at all

<details>
<summary>flutter doctor -v</summary>

```bash
[✓] Flutter (Channel stable, 1.22.6, on macOS 11.2 20D64 darwin-x64, locale en-GB)
    • Flutter version 1.22.6 at /Users/tahatesser/Code/flutter_stable
    • Framework revision 9b2d32b605 (3 weeks ago), 2021-01-22 14:36:39 -0800
    • Engine revision 2f0af37152
    • Dart version 2.10.5
[✗] Android toolchain - develop for Android devices
    ✗ Unable to locate Android SDK.
      Install Android Studio from: https://developer.android.com/studio/index.html
      On first launch it will assist you in installing the Android SDK components.
      (or visit https://flutter.dev/docs/get-started/install/macos#android-setup for detailed instructions).
      If the Android SDK has been installed to a custom location, set ANDROID_SDK_ROOT to that location.
      You may also want to add it to your PATH environment variable.
[!] Xcode - develop for iOS and macOS (Xcode 12.4)
    • Xcode at /Volumes/Extreme/Xcode.app/Contents/Developer
    • Xcode 12.4, Build version 12D4e
    ✗ CocoaPods not installed.
        CocoaPods is used to retrieve the iOS and macOS platform side's plugin code that responds to your plugin usage on the Dart side.
        Without CocoaPods, plugins will not work on iOS or macOS.
        For more info, see https://flutter.dev/platform-plugins
      To install:
        sudo gem install cocoapods
[!] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
[✓] VS Code (version 1.53.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.19.0
[✓] Connected device (1 available)
    • Taha’s iPhone (mobile) • 00008020-001059882212002E • ios • iOS 14.4
! Doctor found issues in 3 categories.
```

</details>


Can you please try to reproduce the issue with the following steps
1. clone  plugins repo `git clone  https://github.com/flutter/plugins.git`
2. Run `plugins/packages/image_picker/image_picker/example` on your device

Thank you

**Comment by CoderJava** [comment]
> Hi @CoderJava
> Tried to reproduce it again using `0.7.0-nullsafety`, with flash off, fresh install, no green at all on `iOS 14.4` device
> 
> flutter doctor -v
> Can you please try to reproduce the issue with the following steps
> 
> 1. clone  plugins repo `git clone  https://github.com/flutter/plugins.git`
> 2. Run `plugins/packages/image_picker/image_picker/example` on your device
> 
> Thank you

Ok. I will try it.

**Comment by CoderJava** [comment]
> Hi @CoderJava
> Tried to reproduce it again using `0.7.0-nullsafety`, with flash off, fresh install, no green at all on `iOS 14.4` device
> 
> flutter doctor -v
> Can you please try to reproduce the issue with the following steps
> 
> 1. clone  plugins repo `git clone  https://github.com/flutter/plugins.git`
> 2. Run `plugins/packages/image_picker/image_picker/example` on your device
> 
> Thank you

Hi @TahaTesser  
I have following your steps. But still the same error.
<details>
  <summary>flutter doctor -v</summary>
  
  ```
    [✓] Flutter (Channel beta, 1.25.0-8.1.pre, on Mac OS X 10.15.7 19H2 darwin-x64, locale en-EC)
    • Flutter version 1.25.0-8.1.pre at /Users/yudisetiawan/Downloads/flutter
    • Framework revision 8f89f6505b (9 weeks ago), 2020-12-15 15:07:52 -0800
    • Engine revision 92ae191c17
    • Dart version 2.12.0 (build 2.12.0-133.2.beta)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /Users/yudisetiawan/Library/Android/sdk
    • Platform android-30, build-tools 30.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.0.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.0.1, Build version 12A7300
    • CocoaPods version 1.10.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] IntelliJ IDEA Community Edition (version 2020.2.3)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.53.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.17.0

[✓] Connected device (2 available)
    • Yudi’s iPhone (mobile) • 1b7890540306c7d8155bacffabc03043d4c28bf9 • ios            • iOS 14.0.1
    • Chrome (web)           • chrome                                   • web-javascript • Google Chrome 88.0.4324.150

• No issues found!
  ```
</details>

By the way I got the log just like this.
<details>
  <summary>log</summary>

   ```
   Flutter run key commands.
r Hot reload. 🔥🔥🔥
R Hot restart.
h Repeat this help message.
d Detach (terminate "flutter run" but leave application running).
c Clear the screen
q Quit (terminate the application on the device).
An Observatory debugger and profiler on Yudi’s iPhone is available at: http://127.0.0.1:63557/5jsJ927SaX4=/

Running with unsound null safety
For more information see https://dart.dev/null-safety/unsound-null-safety
[Camera] Failed to read exposureBiasesByMode dictionary: Error Domain=NSCocoaErrorDomain Code=4864 "*** -[NSKeyedUnarchiver _initForReadingFromData:error:throwLegacyExceptions:]: data is NULL" UserInfo={NSDebugDescription=*** -[NSKeyedUnarchiver _initForReadingFromData:error:throwLegacyExceptions:]: data is NULL}
[Runner] findWriterForType:128: unsupported file format 'public.heic'
[Runner] findWriterForType:128: unsupported file format 'public.heic'
   ```
</details>

**Comment by CoderJava** [comment]
I have fixed this issue. The problem is from iOS 14. I'm just updated to iOS 14.4.1 and the problem is solved.
